src/java.base/share/classes/jdk/internal/util/DecimalDigits.java

```diff
@@ -70,6 +70,21 @@ public final class DecimalDigits implements Digits {
             0x3039, 0x3139, 0x3239, 0x3339, 0x3439, 0x3539, 0x3639, 0x3739, 0x3839, 0x3939
     };
 
+    @Stable
+    private static final int[] DIGITS_K;
+
+    static {
+        int[] digits_k = new int[1000];
```

Comment:
I'm very skeptical that adding a 1000 element lookup table for  is worthwhile. That's a lookup table that'd span many cache lines, with plenty of cache misses. And even _if_ it is to be considered it should be split out and considered in isolation from this PR.

What does the 2, 1 or 0 value you put in the lowest byte signify?

Reply:
DIGITS_K has been removed